framework/wazuh/rbac/orm.py

```diff
@@ -1368,86 +1486,81 @@ def get_policies(self):
+            Flag that indicates if the policy ID can be less than MAX_ID_RESERVED.
 
         Returns
         -------
-        True -> Success | Invalid policy | Missing key (actions, resources, effect) or invalid policy (regex)
+        Union[bool, int]
+            True if the policy was added successfully or a SecurityError code if the operation failed.
         """
         try:
-            if policy is not None and not json_validator(policy):
+            if policy is not None and not isinstance(policy, dict):
                 return SecurityError.ALREADY_EXIST
             if policy is None or len(policy) != 3:
                 return SecurityError.INVALID
-            # To add a policy it must have the keys actions, resources, effect
-            if 'actions' in policy and 'resources' in policy:
-                if 'effect' in policy:
-                    # The keys actions and resources must be lists and the key effect must be str
-                    if isinstance(policy['actions'], list) and isinstance(policy['resources'], list) \
-                            and isinstance(policy['effect'], str):
-                        for action in policy['actions']:
-                            if not re.match(self.action_regex, action):
-                                return SecurityError.INVALID
-                        for resource in policy['resources']:
-                            if not all(re.match(self.resource_regex, res) for res in resource.split('&')):
-                                return SecurityError.INVALID
-
-                        try:
-                            if not check_default:
-                                policies = sorted([p.id for p in self.get_policies()]) or [0]
-                                policy_id = policy_id or max(filter(lambda x: not (x > cloud_reserved_range),
-                                                                    policies)) + 1
-
-                            elif check_default and \
-                                    self.session.query(Policies).order_by(desc(Policies.id)
-                                                                          ).limit(1).scalar().id < max_id_reserved:
-                                policy_id = max_id_reserved + 1
-
-                        except (TypeError, AttributeError):
-                            pass
-                        self.session.add(Policies(name=name, policy=json.dumps(policy), policy_id=policy_id,
-                                                  created_at=created_at))
-                        self.session.commit()
-                        return True
-                    else:
+            # To add a policy, its body must have the `actions`, `resources`, and `effect` keys;
+            # and the values must be instances of `list`, `list` and `str`
+            if all({isinstance(policy.get('actions'), list), isinstance(policy.get('resources'), list),
+                    isinstance(policy.get('effect'), str)}):
```

Comment:
```suggestion
            if all([isinstance(policy.get(attribute), attr_type) for attribute, attr_type in self.policy_attributes.items()]):
```

Declaring `policy_attributes` above as class attribute:

```python
policy_attributes = {
    'actions': list,
    'resources': list,
    'effect': str
}
```
I believe this is a much cleaner and scalable way.

Reply:
Agree, changed in the last commit.